exercises/palindrome-products/palindrome-products.spec.js

```diff
@@ -1,69 +1,125 @@
 
-    expect(smallest.value).toEqual(121);
-    expect(smallest.factors).toEqual([11, 11]);
+    expect(smallest.value).toEqual(expected.value);
+    expect(sortFactors(smallest.factors)).toEqual(expected.factors);
   });
 
   xtest('largest palindrome from triple digit factors', () => {
-    const palindromes = generatePalindrome({ maxFactor: 999, minFactor: 100 });
+    const palindromes = Palindromes.generate({ maxFactor: 999, minFactor: 100 });
     const largest = palindromes.largest;
+    const expected = {value: 906609, factors: [[913, 993]]};
 
-    expect(largest.value).toEqual(906609);
-    expect(largest.factors).toEqual([913, 993]);
+    expect(largest.value).toEqual(expected.value);
+    expect(sortFactors(largest.factors)).toEqual(expected.factors);
   });
 
-  xtest('smallest palindrome from triple digit factors', () => {
-    const palindromes = generatePalindrome({ maxFactor: 999, minFactor: 100 });
+  xtest('smallest palindrome from four digit factors', () => {
+    const palindromes = Palindromes.generate({ maxFactor: 9999, minFactor: 1000 });
+    const smallest = palindromes.smallest;
+    const expected = {value: 1002001, factors: [[1001, 1001]]};
+
+    expect(smallest.value).toEqual(expected.value);
+    expect(sortFactors(smallest.factors)).toEqual(expected.factors);
+  });
+
+  xtest('largest palindrome from four digit factors', () => {
+    const palindromes = Palindromes.generate({ maxFactor: 9999, minFactor: 1000 });
+    const largest = palindromes.largest;
+    const expected = {value: 99000099, factors: [[9901, 9999]]};
+
+    expect(largest.value).toEqual(expected.value);
+    expect(sortFactors(largest.factors)).toEqual(expected.factors);
+  });
+
+  xtest('empty result for smallest if no palindrome in range', () => {
+    const palindromes = Palindromes.generate({ maxFactor: 1003, minFactor: 1002 });
     const smallest = palindromes.smallest;
 
-    expect(smallest.value).toEqual(10201);
-    expect(smallest.factors).toEqual([101, 101]);
+    expect(smallest.value).toBe(null);
+    expect(smallest.factors).toEqual([]);
+  });
+
+  xtest('empty result for largest if no palindrome in range', () => {
+    const palindromes = Palindromes.generate({ maxFactor: 15, minFactor: 15 });
+    const largest = palindromes.largest;
+
+    expect(largest.value).toBe(null);
+    expect(largest.factors).toEqual([]);
+  });
+
+  xtest('error for smallest if min is more than max', () => {
+    expect(() => {
+      const palindromes = Palindromes.generate({ maxFactor: 1, minFactor: 10000 });
+      palindromes.smallest;
+    }).toThrow(new Error('min must be <= max'));
+  });
+
+  xtest('error for largest if min is more than max', () => {
+    expect(() => {
+      const palindromes = Palindromes.generate({ maxFactor: 1, minFactor: 2 });
+      palindromes.largest;
+    }).toThrow(new Error('min must be <= max'));
   });
 });
 
-function numericalArraysMatch(a, b) {
+function factorsMatch(a, b) {
```

Comment:
I don't see this function used anywhere? Am I missing something?

Reply:
Nope it's not.

I had deleted it, must've crept back in during my rebasing. I will remove it.